android/src/main/java/com/klarna/mobile/sdk/reactnative/signin/KlarnaSignInModuleImp.java

```diff
@@ -0,0 +1,116 @@
+package com.klarna.mobile.sdk.reactnative.signin;
+
+import static com.klarna.mobile.sdk.reactnative.common.util.ParserUtil.gson;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeMap;
+import com.klarna.mobile.sdk.KlarnaMobileSDKError;
+import com.klarna.mobile.sdk.api.KlarnaEnvironment;
+import com.klarna.mobile.sdk.api.KlarnaEventHandler;
+import com.klarna.mobile.sdk.api.KlarnaLoggingLevel;
+import com.klarna.mobile.sdk.api.KlarnaMobileSDKCommon;
+import com.klarna.mobile.sdk.api.KlarnaProductEvent;
+import com.klarna.mobile.sdk.api.KlarnaRegion;
+import com.klarna.mobile.sdk.api.component.KlarnaComponent;
+import com.klarna.mobile.sdk.api.signin.KlarnaSignInEvent;
+import com.klarna.mobile.sdk.api.signin.KlarnaSignInSDK;
+import com.klarna.mobile.sdk.api.signin.model.KlarnaSignInToken;
+import com.klarna.mobile.sdk.reactnative.common.util.ArgumentsUtil;
+import com.klarna.mobile.sdk.reactnative.common.util.ParserUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import okhttp3.internal.Util;
+
+public class KlarnaSignInModuleImp implements KlarnaEventHandler {
+
+    private static final String PARAM_NAME_ACTION = "action";
+    private static final String PARAM_NAME_KLARNA_MESSAGE_EVENT = "klarnaMessageEvent";
+    private static final String PARAM_NAME_PARAMS = "params";
+
+    public Promise signInPromise;
+
+    public static final String NAME = "RNKlarnaSignIn";
+
+    private final ReactApplicationContext reactAppContext;
+    private KlarnaSignInSDK signInSDK;
```

Comment:
Having a single reference to an instance here would cause issues for multiple instances created from RN layer, module either needs to store multiple references to these instances for create instances per method calls